pkg/spacelift/spacelift_stack_processor.go

```diff
@@ -366,7 +367,21 @@ func TransformStackConfigToSpaceliftStacks(
 						componentInheritance = i.([]string)
 					}
 
+					baseComponentName := ""
+					if baseComponent, baseComponentExist := componentMap["component"]; baseComponentExist {
+						baseComponentName = baseComponent.(string)
+					}
+
+					if componentMetadata, componentMetadataExists := componentMap["metadata"].(map[interface{}]interface{}); componentMetadataExists {
+						if componentMetadataComponent, componentMetadataComponentExists := componentMetadata["component"].(string); componentMetadataComponentExists {
```

Comment:
This is pretty hard to grok without reading quite a few times. Maybe add a comment on what those nested if blocks are doing? Or consider refactoring the check into a small function with a descriptive name?

Reply:
I made the var names long enough to be able to read and understand what they mean :) 
(it's difficult to comment every line of code).
Anyway, will update